src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java

```diff
@@ -0,0 +1,214 @@
+        if (car != null) {
+            car.setDrivers(getDriversForCar(car.getId()));
+        }
+        return Optional.ofNullable(car);
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String getAllQuery = "SELECT c.id, c.model, m.id AS manufacturer_id,"
+                + " m.name AS manufacturer, m.country "
+                + "AS country "
+                + "FROM cars c JOIN manufacturers m ON c.manufacturer_id = m.id "
+                + "WHERE c.is_deleted = FALSE;";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(getAllQuery)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(getCarFromResultSet(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get a list of cars from cars DB.", e);
+        }
+        for (Car car : cars) {
+            if (car != null) {
+                car.setDrivers(getDriversForCar(car.getId()));
+            }
+        }
+        return cars;
+    }
+
+    @Override
+    public Car update(Car car) {
+        String updateRequest = "UPDATE cars SET model = ?, manufacturer_id = ? "
+                + "WHERE id = ? AND is_deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement updateCarStatement = connection.prepareStatement(updateRequest)) {
+            updateCarStatement.setLong(3, car.getId());
+            updateCarStatement.setString(1, car.getModel());
+            updateCarStatement.setLong(2, car.getManufacturer().getId());
+            updateCarStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't update car in DB " + car,e);
+        }
+        deleteDrivers(car.getId());
+        if (car.getDrivers() != null) {
+            insertDriversToRelationDb(car);
+        }
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long carId) {
+        deleteDrivers(carId);
+        String deleteCarQuery = "UPDATE cars SET is_deleted = TRUE WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement softDeleteCarStatement =
+                        connection.prepareStatement(deleteCarQuery)) {
+            softDeleteCarStatement.setLong(1, carId);
+            int numberOfDeletedRows = softDeleteCarStatement.executeUpdate();
+            return numberOfDeletedRows != 0;
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't delete car with id: " + carId, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        List<Long> carsId = new ArrayList<>();
+        String getAllByDriverQuery = "SELECT * FROM cars_drivers WHERE driver_id = ?";
```

Comment:
Use statement with JOIN here. Do not `use get(id).get()` inside loop.
do not forget `AND is_deleted = FALSE`